reconcile/templating/renderer.py

```diff
@@ -287,6 +288,11 @@ def reconcile(
         gql_no_validation = init_from_config(validate_schemas=False)
 
         for c in get_template_collections():
+            if (
+                self.params.template_collection_name
+                and c.name != self.params.template_collection_name
+            ):
+                continue
```

Comment:
Can we put this logic in gql? To use [$filter](https://github.com/app-sre/qontract-server?tab=readme-ov-file#gql-query-filtering) in gql query, then we can expand filter fields later.

```gql
query TemplateCollection_v1($filter: JSON) {
  template_collection_v1(filter: $filter) {
```

Then pass `variables` to `query` function

```python
    variables = {"filter": {}}
    if template_collection_name:
        variables["filter"]["name"] = template_collection_name
    query(query_func, variables=variables)
```